scrapper.py

```diff
@@ -23,75 +36,310 @@ class IncorrectNumberOfArticlesError(Exception):
+                if article_id >= self.max_articles:
+                    break
 
-    def get_search_urls(self):
+    @staticmethod
+    def get_page(url):
+        """
+        Returns requests page
+        """
+        response = requests.get(url, headers=Crawler.headers)
+        response.encoding = 'utf-8'
+        if response.status_code == 200:
+            page = response.text
+        else:
+            raise BadStatusCode
+        sleep(randint(3, 10))
+        return page
+
+    def process_page(self, url, article_id):
+        """
+        Processes page and get artcle urls
+        """
+        page = self.get_page(url)
+        with open(os.path.join(constants.PAGES_ROOT, f'{article_id}.html'),
+                  'w', encoding='utf-8') as file:
+            file.write(page)
+        soup = BeautifulSoup(page, 'html.parser')
+        article_urls = self._extract_url(soup, self.urls, self.seen_urls)
+        self.urls.extend(article_urls)
+        return soup
+
+
+class CrawlerRecursive(Crawler):
+    """
+    Recursive Crawler implementation
+    """
+
+    def find_articles(self):
+        """
+        Finds articles
+        """
+        article_id = 1
+        while len(self.urls) < self.max_articles and self.seed_urls:
+            seed_url = self.seed_urls.pop()
+            try:
+                soup = self.process_page(seed_url, article_id)
+            except BadStatusCode:
+                continue
+            else:
+                self.seed_urls = self.get_search_urls(soup)
+                article_id += 1
+        self.urls = self.urls[:self.max_articles]
+
+    def get_search_urls(self, soup):
         """
         Returns seed_urls param
         """
-        pass
+        for idx, link in enumerate(soup.find_all('a')):
+            if link.get('href'):
+                if res := re.findall(r'https://www.zvezdaaltaya.ru/.+', link.get('href')):
+                    if res[0] not in self.seen_urls:
+                        self.seed_urls.append(res[0])
+                        self.seen_urls.add(res[0])
+            if idx % 10 == 0:
+                with open(constants.TO_PARSE_URLS, 'w', encoding='utf-8') as file:
+                    file.write('\n'.join(self.seed_urls))
+                with open(constants.SEEN_URLS, 'w', encoding='utf-8') as file:
+                    file.write('\n'.join(self.seen_urls))
+        return self.seed_urls
 
 
 class ArticleParser:
     """
     ArticleParser implementation
     """
     def __init__(self, full_url: str, article_id: int):
-        pass
+        self.article = Article(full_url, article_id)
 
     def _fill_article_with_text(self, article_soup):
-        pass
+        for paragraph in article_soup.find_all('p')[:-3]:
+            self.article.text += paragraph.text + ' '
 
     def _fill_article_with_meta_information(self, article_soup):
-        pass
+        self.article.title = re.sub(r' - Звезда Алтая', '', article_soup.find('title').text)
```

Comment:
the page contains the title that does not require any fragile manipulations like these. find the better way

Reply:
updated it